recreated header/footer components

Willian1661 opened the pull request:
close #4

Willian1661 committed:
recreated header/footer components

eerison commented on client/components/footer.tsx:
```diff
@@ -1,30 +1,30 @@
+                  <span className="sr-only">Facebook</span>
+              </Link>
+              <Link href="/" className="text-gray-500 hover:text-purple-700">
+                  <svg className="w-6 h-6" fill="currentColor" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M12.315 2c2.43 0 2.784.013 3.808.06 1.064.049 1.791.218 2.427.465a4.902 4.902 0 011.772 1.153 4.902 4.902 0 011.153 1.772c.247.636.416 1.363.465 2.427.048 1.067.06 1.407.06 4.123v.08c0 2.643-.012 2.987-.06 4.043-.049 1.064-.218 1.791-.465 2.427a4.902 4.902 0 01-1.153  ...
```
what is this inside of `d` attribute?

Willian1661 replied:
this `d` attribute it "makes" the icons, it's like lithe pieces of the svg that creates the icon
i didn't removed because i would have to add the tablericons package, that i didn't istalled yet

Willian1661 replied:
and cause of this i wanted to create another pr to remove all these svg's tags

eerison replied:
Ok

Willian1661 replied:
or i can just use this pr to add the icons ready? is much thing?

eerison commented on client/components/footer.tsx:
```diff
@@ -1,30 +1,30 @@
-            >
-              View on GitHub
-            </a>
+    <div className="w-full mx-auto container md:p-8 p-6 md:flex md:items-center md:justify-between">
```
cloud you use `@apply` from tailwind?

Willian1661 replied:
create a "template" on tailwind config?

eerison replied:
Create one css per component

Willian1661 replied:
you mean i create a css header with all the styles os header and it's childs? and so on

Willian1661 replied:
ok i created one folder components/ inside of styles and more two css files into it.

eerison commented on client/pages/index.tsx:
```diff
@@ -7,6 +7,7 @@ import { getAllPosts } from '../lib/api'
 import Head from 'next/head'
 import { CMS_NAME } from '../lib/constants'
 import Post from '../interfaces/post'
+import Header from '../components/header'
```
create a test to check if header is called inside of index.tsx

Willian1661 replied:
ok

Willian1661 replied:
hey, i created two test to create `snapshot` of `header & footer`, but i don't know how can i test,**for example:** if header is being called inside of index component.
because index it receives a array of parameters i think, i'm not sure:
![Screenshot from 2023-03-28 17-39-04](https://user-images.githubusercontent.com/106893704/228361320-271db02e-6cd0-4746-b3b6-d96c76c6e71a.png)

and it returns others components, i tried mock the "array of parameters" but istill nothing.

eerison replied:
Header component should be inside of layout, shouldn't it ?

Willian1661 replied:
I thought would be, but next use just footer inside of layout and header inside of the posts  separatly.

Willian1661 replied:
Layout use a main tag to place the components inside of layout, maybe I could try to use header inside of this main tag

eerison replied:
It is not defined by next,  it's defined by blog template,  and it can be changed  ;)

Willian1661 replied:
i got it, i moved the header component to layout with footer

eerison commented on client/components/header.tsx:
```diff
@@ -2,12 +2,39 @@ import Link from 'next/link'
+      <div className="container flex flex-wrap items-center justify-between mx-auto my-2">
+        <Link href="/" className="flex items-center">
+          <img src="https://flowbite.com/docs/images/logo.svg" className="h-6 mr-3 sm:h-9" alt="Flowbite Logo" />
+          <span className="self-center text-xl font-semibold whitespace-nowrap">SeuJob</span>
```
let keep just `job` for now ;)

Willian1661 replied:
ok then

Willian1661 replied:
it's now

eerison commented on client/pages/index.tsx:
```diff
@@ -7,6 +7,7 @@ import { getAllPosts } from '../lib/api'
 import Head from 'next/head'
```
post here a screenshot how, it's in desktop and mobile :)

Willian1661 replied:
i didn't created resolution for mobile screen, just md screen

<div>
<img src="https://user-images.githubusercontent.com/106893704/226742925-24e1136b-4192-4947-bc30-1a5e0a4ff5e0.png" width="350"></img>
<img src="https://user-images.githubusercontent.com/106893704/226742997-ce2417d7-a215-48db-8f1c-edd72d4b15b4.png" width="350"></img>
</div>

eerison replied:
Remove this messa/block on top of navbar

eerison replied:
Icons centralized in the footer 

How can I see the menu,  like about, post job ?

Willian1661 replied:
the menu and stuff i still have to fix,is breaked so header as footer.

eerison replied:
Then it isn't done yet, is it?

Willian1661 replied:
no isn't

Willian1661 replied:
header and footer are also responsive now

Willian1661 replied:
i changed them both, for now on the header it has a drop down that don't pop up nothing, because i was thinking to add this functionality on another pr to don't overload this pr

eerison replied:
Ok could you post a new screenshot :)

Willian1661 replied:
ok :p

Willian1661 replied:
these are header/footer on desktop size(1440):

![Screenshot from 2023-04-17 17-54-21](https://user-images.githubusercontent.com/106893704/232608559-979209b5-5593-4a70-a49a-a1e95657eeac.png)
![Screenshot from 2023-04-17 17-54-10](https://user-images.githubusercontent.com/106893704/232608603-af6811a9-8e14-4f60-bf8a-2569a2617865.png)

these are header/footer on mobel size(320):
<div>
<img src="https://user-images.githubusercontent.com/106893704/232608613-f05a9e3c-c9df-40e6-825c-8206d54e5a28.png" width="200px"></img>
<img src="https://user-images.githubusercontent.com/106893704/232608845-8fb7ecc0-643b-4a6e-9720-01b8270eeab7.png" width="200px"></img>
</div>

Willian1661 replied:
@eerison about the footer, the social midia i left it like this just to be not empty, when add tabler into project i change

Willian1661 replied:
@eerison when you get time for review, change the disk this forro is playing too much :p

Willian1661 committed:
- adjusted header and footer component
- created new tests for them and added new babel config
- moved header component for layout component and removed it from slug and index
- created styles for both

Willian1661 commented:
this commit it's bigger because i added new packages for the tests that i finally discovered how to create them to support tsx components :p

eerison commented on client/_tests_/components/__snapshots__/footer.test.tsx.snap:
```diff
@@ -0,0 +1,114 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`footer test is redendring properly footer component  1`] = `
+<footer
```
Isn't possible just check if the componet was called? Otherwise this test gonna fail for every small change  :/

Willian1661 replied:
yea could be, you saying now i think is better to check if it was called then if it contains something

Willian1661 replied:
but it returns a html component, why don't use snapshot?
ok it will fail every change, but if is a html component how i will know is right?

Willian1661 replied:
do you think is better to create a snapshot for layout and just check if header and footer was called?
because footer and header literally repeats them self's inside layout snapshot

eerison replied:
use a snap shot for a small component is fine, but a huge component is not a good idea.
for this reason should be good check only if it was called, and not the html output.

for small component I guess it make sense, but for big output, it doesn't looks good.

Willian1661 replied:
ok got it

Willian1661 replied:
hey i thinking about to drop just 1 test, stay with layout's tests to check if the others components are right, what u think?

Willian1661 replied:
hey do you think add one more issue int this pr?? 
to add tabler to remove that svg codes

eerison replied:
It should be in other pr

Willian1661 replied:
ok then

Willian1661 replied:
the tests are done.

Willian1661 committed:
- adjusted component's tests
- moved _test_ directory into components
- removed old tests

eerison commented on client/styles/components/footer.css:
```diff
@@ -0,0 +1,15 @@
+    @apply bg-neutral-50 border-t border-neutral-200
+}
+
+.footer-container{
```
it's doing more then one thing

it should be juts footer, or container.

Willian1661 replied:
i wanted to be refereed about footer's container not be a footer and also a container

eerison commented on client/styles/components/header.css:
```diff
@@ -0,0 +1,25 @@
+  @apply bg-white border-gray-200 px-2 sm:px-4 py-2.5 rounded;
+}
+
+.nav-container {
```
same here.

create a container class, to use in different context.

Willian1661 replied:
hmm make sense, i'm thinking about use these APPLY just for this context and not for many others, ok got it

Willian1661 replied:
@eerison is it wrong the way i named the container class ? 
because i meant to have two different containers but still being a container
or do you think is best practice a create a container with mutual features and add other classes into my container class ?

eerison replied:
maybe rename `.Container-2` to `.container-full`, and I guess those containers should be inside something like

`basic.css`, because it is not related with footer.

Willian1661 replied:
for the cont-1 i named `container-flex`, because on the css line it uses more flex attribuites
and the same thing for the cont-2 i named `container-full`

eerison commented on client/styles/components/header.css:
```diff
@@ -0,0 +1,25 @@
+  @apply container flex flex-wrap items-center justify-between mx-auto my-2;
+}
+
+.nav-ul {
```
- Maybe you could use more explicitly name like `navbar`.
- think in other name to this `ul` or just remove it (you can assume the all navbar will be inside of a `ul`).

Willian1661 replied:
this one i think it's the same situation as `nav-container` and `footer-container`, because i'm using this for a `ul`, that belongs to a `navbar`, but could use for another `ul` i think.

eerison commented on client/styles/components/header.css:
```diff
@@ -0,0 +1,25 @@
+.nav-ul {
+  @apply flex flex-col p-4 mt-4 border border-gray-100 rounded-lg bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 md:bg-white;
+}
+.nav-main-li{
```
maybe `navbar-item`?

Willian1661 replied:
could be, but this is just a specific type of item you feel me? with different details
 is `navbar-main-item` bad?

eerison replied:
it is fine

eerison commented on client/styles/components/header.css:
```diff
@@ -0,0 +1,25 @@
+.nav-li{
+  @apply block w-full py-2 pl-3 pr-4 text-gray-700 hover:bg-gray-100 md:hover:bg-transparent md:hover:text-blue-700 md:p-0 inline-block p-4 border-b-2 border-transparent hover:text-blue-700 hover:border-blue-600;
+}
+.nav-dropDown-li{
```
you can remove dropdow, let's make it simple for now.

Willian1661 replied:
ok then

Willian1661 replied:
i was thinking about create another pr to make this dropdown works, but can be later on

eerison commented on client/styles/index.css:
```diff
@@ -1,3 +1,6 @@
 @tailwind base;
```
You can remove this, because you are importing in all files.
or remove for the others files and keep just here.

Willian1661 replied:
yea i removed from the the other files, it is only in index now

Willian1661 committed:
- adjusted header/footer component and their css files

Willian1661 committed:
- added basic.css
- changed header/footer tests
- edited components

Willian1661 committed:
updated footer's div space

eerison commented:
Just post a Screenshot in desktop and mobile version  :)

Willian1661 commented:
> Just post a Screenshot in desktop and mobile version :)

i already posted, is up there https://github.com/shield-wall/job/pull/8#discussion_r1169261126

eerison commented:
> > Just post a Screenshot in desktop and mobile version :)
> 
> i already posted, is up there [#8 (comment)](https://github.com/shield-wall/job/pull/8#discussion_r1169261126)

Ok, why is there a broken image?

Willian1661 commented:
> > > Just post a Screenshot in desktop and mobile version :)
> > 
> > 
> > i already posted, is up there [#8 (comment)](https://github.com/shield-wall/job/pull/8#discussion_r1169261126)
> 
> Ok, why is there a broken image?

beucause that was the flowbitte logo and i removed to have nothing from outside and for not have a random logo

eerison commented on client/components/_tests_/footer.test.tsx:
```diff
@@ -0,0 +1,11 @@
+    it('is redendring properly footer component ', () => {
+        const tree = renderer.create(
+            <Footer />
+        ).toJSON()
```
why do you convert to json?

Willian1661 replied:
to be more easy to read the test code

Willian1661 replied:
because, as i remember when i don't convert them `toJSON` they stay not formatted

eerison commented on client/package.json:
```diff
@@ -20,13 +20,21 @@
   },
   "devDependencies": {
-    "@types/jest": "^29.4.0",
+    "@babel/core": "^7.21.3",
```
if those libs are necessary only for jest, then you can add it in dev dependencies

Willian1661 replied:
what do you mean? but they are already in devDependencies

eerison replied:
Sorry,  that's true

eerison commented on client/components/_tests_/footer.test.tsx:
```diff
@@ -0,0 +1,11 @@
+        const tree = renderer.create(
+            <Footer />
+        ).toJSON()
+        expect(tree).toMatchSnapshot();
```
I guess you could just check if the `id` links exist inside of footer.

you don't need to check the whole structure, because this test will be so fragile  :/

Willian1661 replied:
![Screenshot from 2023-04-24 18-57-05](https://user-images.githubusercontent.com/106893704/234125471-d98b5b3f-4876-4956-8a53-ddf7fa754536.png)

you mean, i create a `ID` for footer and test if this `ID` is being returned?

eerison replied:
Yep, I think, it's fine

Willian1661 replied:
Ok then, i'm getting now that isn't so good use snapshot for these components(or every component), every time i update i forgot to to update the snap as well and then i have to reset my commit and force a new one :´)

Willian1661 replied:
dude what a hit on my head XD, is being harder than i thought. In my head it was a lot more easier

Willian1661 committed:
- recreated new tests for header and footer
- added testid for both
- added new package
- edited jest.config for the tests

eerison commented on client/components/footer.tsx:
```diff
@@ -1,32 +1,30 @@
+    <footer className="footer" data-testid="footer">
+      <div className="container-full">
+        <span className="text-md text-gray-500">© 2023
+          <a href="/" className="hover:underline">
```
`hover:underline` ?

why do you create a class like this?

Willian1661 replied:
i needed just one attribute, so just called one attribute and i didn't created like the others with @apply, is just a attribute a called

eerison commented on client/components/_tests_/footer.test.tsx:
```diff
@@ -0,0 +1,11 @@
+describe("footer test", () => {
+    it('is redendring properly footer component ', () => {
+        const { getByTestId } = render(<Footer />);
+        const elementId = getByTestId('footer');
```
you should check if the links id are inside of footer, and not footer.

Willian1661 replied:
what link ? footer has no link inside of it, layout has a link of footer, when you said id i was thinking to check if footer is creating the component footer and the same for header

eerison commented on client/components/_tests_/header.test.tsx:
```diff
@@ -0,0 +1,11 @@
+describe("Header test", () => {
+  it("is rendering properly header component", () => {
+   const { getByTestId } = render(<Header />);
+   const elementId = getByTestId('header');
```
same here

Willian1661 replied:
i removed all this, i'm getting the links by the container("dom") with querySelector and i'm using it because by the way i did it there's no method `getElementByid()`.

Willian1661 replied:
@eerison i changed a few things after i showed you the new tests, how it is ?

eerison commented on client/components/footer.tsx:
```diff
@@ -1,32 +1,30 @@
-              View on GitHub
-            </a>
-          </div>
+    <footer className="footer" data-testid="footer">
```
`data-testid="footer"` ?

Willian1661 replied:
this testid was to get the footer's id , was what thought had to do :P

Willian1661 replied:
i removed this, i'm trying other way to get the `ids`

Willian1661 committed:
updat tests...
- created new tests for header and footer component
- added id to all links and first tags
- tests to check links

eerison commented on client/components/_tests_/header.test.tsx:
```diff
@@ -0,0 +1,25 @@
+    expect(header.getAttribute('id')).toBe('header')
+  })
+
+  it.each(links)("is checking header's links path exists", (a) => {
```
Rename a to link

Willian1661 replied:
i named it as `links` 'cause i get all the links at once, you thinking is more appropriated name it as `link` even if it has more then one link?

Willian1661 replied:
done

eerison commented on client/package.json:
```diff
@@ -3,7 +3,7 @@
     "dev": "next",
     "test": "jest --coverage",
-    "lint": "eslint --ext .js,.ts .",
+    "lint": "npx eslint components/**",
```
You need to check the whole project 

npx eslint


Not just components

Willian1661 replied:
all the structure with same pattern right?

Willian1661 replied:
done

Willian1661 committed:
ran eslint and updated tests

Willian1661 committed:
changed 'a' variable to 'link'
